public/pages/MonitorDetails/containers/Triggers/Triggers.js

```diff
@@ -60,11 +60,8 @@ export default class Triggers extends Component {
       }),
       {}
     );
-    const shouldKeepTrigger = (trigger) =>
-      !triggersToDelete[
-        isAggregationMonitor ? trigger.aggregation_trigger.name : trigger.traditional_trigger.name
-      ];
-    const updatedTriggers = monitor.triggers.filter(shouldKeepTrigger);
+    const shouldKeepTrigger = (trigger) => !triggersToDelete[trigger.name];
+    const updatedTriggers = this.getUnwrappedTriggers(monitor).filter(shouldKeepTrigger);
```

Comment:
Did you test this with both Trigger types? Passing in the unwrapped Triggers to `updateMonitor` might fail because the API needs the Trigger type to parse it correctly.

Reply:
Good callout! I've been able to successfully update aggregation and traditional visual graph triggers.

Reply:
I'm pretty sure if the trigger type is removed when defining a Trigger in the Monitor definition, it will parse it as a Traditional Trigger (this was done to be backwards compatible with the old format where there was no trigger type). I think we might want to keep the filter type here.

So the update Monitor API call might be passing but I'm not sure if it's doing what we expect.

Reply:
I'll merge in the change and confirm if this is happening or not, if it is, I'll push an update to the dev branch.

Reply:
Looks like the update Monitor API call fails when a Trigger is deleted due to the aforementioned issue.

Reply:
Thank you for deep diving this!